### PR TITLE
[WIP] Implement IDbContextFactory to work around aspnet/EntityFramework#8164

### PR DESCRIPTION
**Pull request by bricelam**
@Eilon @ajcvickers @DamianEdwards @divega @danroth27

### CODE REVIEW

**Comment by bricelam** [comment]
I haven't tested this yet; I just wanted to get it out there for early review.

**Comment by Eilon** [comment]
Only thing I don't know is whether there ought to be a comment on there. I'd add one only if it provided real value, as opposed to stating the obvious.

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Data/ApplicationDbContextFactory.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using Microsoft.Extensions.DependencyInjection;
+using Company.WebApplication1.Models;
```
Comment (bricelam) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Data/ApplicationDbContextFactory.cs]: Will remove this `using`.

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Data/ApplicationDbContextFactory.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+namespace Company.WebApplication1.Data
+{
+    public class ApplicationDbContextFactory : IDbContextFactory<ApplicationDbContext>
+    {
+        public ApplicationDbContext Create(DbContextFactoryOptions options)
+            => Program.BuildWebHost(Array.Empty<string>())
```
Comment (DamianEdwards) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Data/ApplicationDbContextFactory.cs]: Move the `=>` to the end of the declaring line

Comment (bricelam) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Data/ApplicationDbContextFactory.cs]: ...and format it like this?
```C#
public ApplicationDbContext Create(DbContextFactoryOptions options) => Program
    .BuildWebHost(Array.Empty<string>())
    .Services
    .GetRequiredService<ApplicationDbContext>();
```

Comment (bricelam) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Data/ApplicationDbContextFactory.cs]: (to match [Program.BuildWebHost](https://github.com/dotnet/templating/blob/rel/vs2017/post-rtw/template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Program.cs#L20-L23))

Comment (DamianEdwards) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Data/ApplicationDbContextFactory.cs]: Egad, no, like this:
``` c#
public ApplicationDbContext Create(DbContextFactoryOptions options) =>
    Program.BuildWebHost(Array.Empty<string>()).Services.GetRequiredService<ApplicationDbContext>();
```

Comment (Eilon) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Data/ApplicationDbContextFactory.cs]: Tsk tsk @danroth27 .

**Comment by bricelam** [comment]
Updated

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Data/ApplicationDbContextFactory.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore.Infrastructure;
```
Comment (Eilon) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Data/ApplicationDbContextFactory.cs]: Kind of an odd namespace for this apparently very useful type... but I guess that's not relevant to this PR...

**Comment by bricelam** [comment]
@mlorbetske I wish you would have waited to merge this; aspnet/EntityFramework#8314 will break this...

Oh well, I'll send a new PR when it does.

**Comment by mlorbetske** [comment]
Sorry, got the go ahead from @ajcvickers & @DamianEdwards

**Comment by DamianEdwards** [comment]
Ahem, I believe I asked you to check with @bricelam first 🤔

**Comment by bricelam** [comment]
No biggie. Sent #705

**Comment by mlorbetske** [comment]
@DamianEdwards I tried to but got redirected. Anyway, it seems to be working out.

**Pull request merged**